woocommerce/woocommerce-ios
# Update "friendly string" logic

Lot's of updates to "friendly string" logic in this PR. A picture is worth a thousand words:

![3](https://user-images.githubusercontent.com/154014/52086347-42449f80-256c-11e9-8c51-3346d02e83df.png)

![3a](https://user-images.githubusercontent.com/154014/52086365-4b357100-256c-11e9-86f8-5eb1abf9fd2e.png)

## Discussion

In general, I wanted to improve upon the existing numerical recipe found the friendly string logic we extended `Double` with. As it stood, the existing logic could not be localized easily and was problematic to use with currency values.

Some of the specific updates here include:
* Renamed the`Double+Woo.swift` extension's `friendlyString()` method to `humanReadableString()`
* Refactored the existing logic in`Double.humanReadableString()` to make use of NumberFormatter for both small and large numbers so we have better i18n support
* Added `humanReadableString()` to the `NSDecimalNumber+Helpers.swift` extension
* Added a new func to `CurrencyFormatter` → `formatHumanReadableAmount()` which will yield a human-readable string with the ability to both round and not round small numbers to the nearest integer
* In Top Earners, Period Stats, and the Order List screens, I updated all of the appropriate "friendly string" calls to use the new `formatHumanReadableAmount()` func in `CurrencyFormatter`
* Added a lot more unit tests

**Note:** even with all these updates, we are still using the currency settings as defined on the store website.

Fixes: #573 

## Testing

* Make sure all the logic here makes sense
* Run the unit tests — verify they are ✅ 
* Build and run the app — verify everything appears as you would expect (focus on the Order List, and Dashboard screens)
* Build and run the app using different languages — verify everything appears as you would expect

----

@mindgraffiti Please give this a good looksie (sorry about the length here, a lot of it is project files, inline doc comments, and unit tests. 😄 )

@astralbodies if you could put some 👀 on this too, that would be grand!
